skimage/draw/draw.py

```diff
@@ -764,3 +764,76 @@ def rectangle(start, end=None, extent=None, shape=None):
     coords = np.meshgrid(*[np.arange(st, en) for st, en in zip(tuple(tl),
                                                                tuple(br))])
     return coords
+
+
+def rectangle_perimeter(start, end=None, extent=None, shape=None, clip=False):
+    """Generate coordinates of pixels that are exactly around a rectangle.
+
+    Parameters
+    ----------
+    start : tuple
+        Origin point of the inner rectangle, e.g., ``(row, column)``.
+    end : tuple
+        End point of the inner rectangle ``(row, column)``.
+        Either `end` or `extent` must be specified.
+    extent : tuple
+        The extent (size) of the inner rectangle.  E.g.,
+        ``(num_rows, num_cols)``.
+        Either `end` or `extent` must be specified.
+    shape : tuple, optional
+        Image shape used to determine the maximum bounds of the output
+        coordinates. This is useful for clipping perimeters that exceed
+        the image size. By default, no clipping is done.
+    clip : bool, optional
+        Whether to clip the perimeter to the provided shape. If this is set
+        to True, the drawn figure will always be a closed polygon with all
+        edges visible.
+
+    Returns
+    -------
+    coords : array of int, shape (Ndim, Npoints)
+        The coordinates of all pixels in the rectangle.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage.draw import rectangle_perimeter
+    >>> img = np.zeros((5, 6), dtype=np.uint8)
+    >>> start = (2, 3)
+    >>> end = (3, 4)
+    >>> rr, cc = rectangle_perimeter(start, end=end, shape=img.shape)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0, 0],
+           [0, 0, 1, 1, 1, 1],
+           [0, 0, 1, 0, 0, 1],
+           [0, 0, 1, 0, 0, 1],
+           [0, 0, 1, 1, 1, 1]], dtype=uint8)
+
+    >>> img = np.zeros((5, 5), dtype=np.uint8)
+    >>> rr, cc = rectangle_perimeter(start, end=(10, 10), shape=img.shape,
+                                     clip=True)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0],
+           [0, 0, 1, 1, 1],
+           [0, 0, 1, 0, 1],
+           [0, 0, 1, 0, 1],
+           [0, 0, 1, 1, 1]], dtype=uint8)
+
+    """
+
+    if extent is not None:
+        end = np.array(start) + np.array(extent)
```

Comment:
Rather `np.asarray` here.

Reply:
Just a side note: I think we have to restrict usage of negative `extent` values. Here and also in `rectangle` function. This could be very confusing.

Reply:
`asarray` ok.

I could add logic like:
```python
        extent = np.asarray(extent)
        if (extent < 0).any():
            raise ValueError("Extent should be positive.")
        end = np.array(start) + extent
```
but then that would duplicate logic like 
```
    tl = np.minimum(start, end)
    br = np.maximum(start, end)
```

Reply:
What I meant here is that negative `extend` values are hard to understand (at least, for myself). For example, `extent (1,1)` returns a perimeter around 1x1 area. But for `extent (-1, -1)` it is not so intuitively to predict what will be the output. A simple solution to this issue would be to disallow negative values of `extent`.
If you think, however, that negative `extent`'s are useful, and it is not confusing at all, we can consider leaving this part as it is, and improving the docstring.

Reply:
Got it. I'll see what negative extents actually do. It is not something I had considered.

Reply:
Negative extents seem to allow you to specify a corner other than the top left one. Could be pretty useful.

```python
from skimage.draw import rectangle_perimeter
img = np.zeros(shape=(6, 6), dtype=int)
rr, cc = rectangle_perimeter(start=(3, 3), extent=(0, 0))
img[rr, cc] = 2
print(img)


img = np.zeros(shape=(6, 6), dtype=int)
rr, cc = rectangle_perimeter(start=(3, 3), extent=(-1, -1))
img[rr, cc] = 2
print(img)

img = np.zeros(shape=(6, 6), dtype=int)
rr, cc = rectangle_perimeter(start=(3, 3), extent=(-1, 2))
img[rr, cc] = 2
print(img)
```